SRC/NETWORK/hget.c

```diff
@@ -528,18 +562,37 @@ int CharIsWhitespace(char ch)
 void ProcessUrl(char* url, byte isRedirection)
 {
     char* pointer;
-    
+
     if(url[0] == '/') {
         if(isRedirection) {
             redirectionUrlIsNewDomainName = 0;
         } else {
             Terminate(strInvParam);
         }
     } else if(StringStartsWith(url, "http://")) {
+		TcpConnectionParameters->remotePort = HTTP_DEFAULT_PORT;
+		TcpConnectionParameters->flags = 0 ; //set flags properly for HTTP in case of redirection...
         if(isRedirection) {
-            redirectionUrlIsNewDomainName = 1;
+			pointer = FindFirstSlash(url+7);
+			if ((pointer)&&(strncmpi(url+7, domainName, (pointer-url-7))))
+				redirectionUrlIsNewDomainName = 1;
+			else
+				redirectionUrlIsNewDomainName = 0;
         }
         strcpy(domainName, url + 7);
+    } else if((bTls)&&(StringStartsWith(url, "https://"))) {
+        if(isRedirection) {
+			pointer = FindFirstSlash(url+8);
+			if ((pointer)&&(strncmpi(url+8, domainName, (pointer-url-8))))
+				redirectionUrlIsNewDomainName = 1;
+			else
+				redirectionUrlIsNewDomainName = 0;
+        }
+        strcpy(domainName, url + 8);
+		bHTTPS = true;
+		TcpConnectionParameters->remotePort = HTTPS_DEFAULT_PORT;
+		TcpConnectionParameters->flags = TcpConnectionParameters->flags | 4 ; //set flags properly for TLS

```

Comment:
Please convert that 8 to a constant, e.g. `TCPFLAGS_USE_TLS`. This way the code reads better and you don't need the comment.

Reply:
Done :)